src/private/strarray.cpp

```diff
@@ -22,30 +22,48 @@
 namespace LibQGit2 {
 namespace internal {
 
-StrArray::StrArray()
-{
-    m_data.count = 0;
-    m_data.strings = NULL;
-}
+static_assert(!std::is_copy_constructible<StrArray>::value, "StrArray must NOT be copy constructible");
+static_assert(std::is_move_constructible<StrArray>::value, "StrArray must be move constructible");
+static_assert(!std::is_copy_assignable<StrArray>::value, "StrArray must NOT be copy assignable");
+static_assert(std::is_move_assignable<StrArray>::value, "StrArray must be move assignable");
 
 StrArray::StrArray(const QList<QByteArray> &list) :
     m_strings(list)
 {
-    m_data.count = 0;
-    m_data.strings = NULL;
+    m_data.count = size_t(m_strings.size());
+
+    if (m_data.count == 0) {
+        return;
+    }
 
-    updateNative();
+    m_data.strings = static_cast<char**>(malloc(m_data.count * sizeof(char*)));
+    for (size_t i = 0; i < m_data.count; ++i) {
+        m_data.strings[i] = const_cast<char*>(m_strings.at(int(i)).data());
+    }
+}
+
+StrArray::StrArray(StrArray &&other) :
+    m_strings(other.m_strings),
+    m_data(other.m_data)
+{
+    other.m_strings.clear();
```

Comment:
When using `std::move()`, it is no longer necessary to `clear()` the strings container.